app/js/components/drugOrderEntry/addForm/AddForm.jsx

```diff
@@ -95,48 +107,52 @@ export class AddForm extends React.Component {
       drugInstructions,
     } = this.state.fields;
 
-    this.setState({
-      draftOrder: {
-        drugName: this.props.drugName,
-        action: this.state.action,
-        dose,
-        dosingUnit,
-        frequency,
-        route,
-        duration,
-        durationUnit,
-        dispensingUnit,
-        dispensingQuantity,
-        reason,
-        previousOrder: this.state.previousOrder,
-        drugInstructions,
-        careSetting: this.props.careSetting.uuid,
-        drug: this.props.drugUuid,
-        orderer: this.props.session.currentProvider.uuid,
-        dosingType: this.state.formType === 'Standard Dosage' ?
-          'org.openmrs.SimpleDosingInstructions' :
-          'org.openmrs.FreeTextDosingInstructions',
-        type: "drugorder",
-        orderNumber: (this.state.action === 'NEW') ? this.props.draftOrders.length :
-          this.props.orderNumber,
-      },
-    }, () => {
-      this.props.addDraftOrder(this.state.draftOrder);
-      if (this.state.draftOrder.action === 'NEW') {
-        this.props.setOrderAction('DRAFT', this.state.orderNumber);
-      } else {
-        this.props.setOrderAction('DRAFT_EDIT', this.props.orderNumber);
-      }
-    });
-    this.props.selectDrugSuccess('');
-    this.clearDrugForms();
-    this.props.clearSearchField();
-    if (this.props.activity === 'DRAFT_ORDER_EDIT') {
-      this.props.setSelectedOrder({
-        currentOrderType: this.props.currentOrderType,
-        selectedOrder: null,
-        activity: null,
+    if (!this.checkIfDrugExist(this.props.drugName)) {
+      this.setState({
+        draftOrder: {
+          drugName: this.props.drugName,
+          action: this.state.action,
+          dose,
+          dosingUnit,
+          frequency,
+          route,
+          duration,
+          durationUnit,
+          dispensingUnit,
+          dispensingQuantity,
+          reason,
+          previousOrder: this.state.previousOrder,
```

Comment:
Hey @dessHub, It'll be nice if you could destructure this.state and this.props.

Reply:
Done.